pkg/fanal/secret/scanner.go

```diff
@@ -335,6 +340,10 @@ func (s Scanner) Scan(args ScanArgs) types.Secret {
 		}
 	}
 
+	censored := make([]byte, len(args.Content))
+	copy(censored, args.Content)
```

Comment:
I'm concerned that we always have two duplicate file contents in memory even though no secrets are found. I understand `censored` is necessary for masking, but unneeded if any secret is found, right? We can delay this copy after one secret at least is found by `sync.Once` or any other approach. But it makes readability less.
Do you think memory usage can be problematic in this case? Unlike other scanning, secret scanning can be applied to almost all files. If you think it is fine, we can go with this approach.

Reply:
Good point. The extra copy of the file will only be in use while that file is being scanned for secrets, but I agree there is no point assigning the memory unless it will be used - I'll go with the `sync.Once` approach, thanks! :+1: